src/server.c

```diff
@@ -3710,15 +3704,6 @@ int processCommand(client *c) {
         return C_ERR;
     }
 
-    /* If we're inside a module blocked context yielding that wants to avoid
-     * processing clients, postpone the command. */
-    if (server.busy_module_yield_flags != BUSY_MODULE_YIELD_NONE &&
-        !(server.busy_module_yield_flags & BUSY_MODULE_YIELD_CLIENTS))
-    {
-        blockPostponeClient(c);
-        return C_OK;
-    }
```

Comment:
I also had the same thought, but I thought it is too implicit assumption that can cause some issues in future implementations or bugs. I also feel we should aim to have as little difference between client calling processCommand and actually calling cmd->proc. the postpone introduced in https://github.com/redis/redis/pull/11193/ is basically a different type of blocking which indeed prevents "command processing" as apposed to the cases handled inside processCommand. I think it is O.K to have modules postpone commands after verifying arity, it is probably what the user would prefer anyway (why should we postpone  something that will never be able to run?)  
I can consider using the CLIENT_PENDING_COMMAND which IMO is a nice candidate to decide on "reprocessing"

Reply:
@oranagra In the end I tried to use the cmd->proc as it seemed like the easiest way to solve the issue. I am not 100% comfortable with it as I have no good way to verify we indeed reprocessing (other then to use the REPROCESSING flag :) ) 
I did run the tests with a combination of the check for cmd->proc and asserting in case we are trying to process a command when this flag is not present and did not get any assert.
Please take a look and share your thoughts

Reply:
the changes LGTM.
i still think it would be nice to be able to know in `call()` that it's not the first time we get there on behalf of the current command. it can be a flag, or maybe even a counter that we can show in CLIENT LIST.
not sure what it would be useful yet, so we can put it aside for now.
maybe @guybe7 has an idea?